hudi-common/src/main/java/org/apache/hudi/common/table/timeline/CompletionTimeQueryView.java

```diff
@@ -126,14 +127,18 @@ public Option<String> getCompletionTime(String baseInstant, String instantTime)
     if (completionTimeOpt.isPresent()) {
       String completionTime = completionTimeOpt.get();
       if (completionTime.length() != baseInstant.length()) {
+        // MDT uses compaction instant time as [delta_instant] + "001".
+        // For example, "20210101000000_001" is the compaction instant time for "20210101000000".
+        // However, in reality, MDT compaction completes in MDT timeline before the deltacommit in data timeline.
+        if (baseInstant.length() - instantTime.length() == 3 && hasCompactionOperationSuffix(baseInstant)) {
+          return completionTimeOpt;
```

Comment:
Do you think we should use directly its own real instant time for MDT compaction instead of suffixing the delta commit ?

Reply:
Done. Updated in `HoodieBackedTableMetadataWriter#compactIfNecessary`